src/layout/FormFields.component.js

```diff
@@ -236,10 +236,19 @@ function createAvatarEditor(fieldBase, d2, valueStore) {
     })
 }
 
-function createVerifyButton(fieldBase, valueStore) {
+function createModalField({ fieldBase, valueStore, onUpdate, d2 }) {
     return Object.assign({}, fieldBase, {
-        component: VerifyEmail,
-        props: { userEmail: valueStore.state['email'] || '' },
+        component: ModalField,
+        props: {
+            onUpdate,
```

Comment:
Why can't we just override `onUpdate` - here? And then set `emailUpdated` within that override? 
I don't think it's necessary to both pass down `onUpdate` and an `setUserEmail` property?